specs/AdaptiveCardsXamarinSpec.md

```diff
@@ -0,0 +1,188 @@
+# AdaptiveCards Xamarin Renderer
+
+## Proposal Link
+
+There is no open issue for a Xamarin.Native package so I'm placing the Xamarin.Forms issue for now 
+https://github.com/microsoft/AdaptiveCards/issues/3255
+
+## Dependencies
+This project depends entirely on the Android and iOS renderer and their API's. This project is a Xamarin bindings library so the only new code to be included are the xamarin transformations for the underlying renderers. For generating the bindings, Xamarin requires the generated binaries (aar file for Android and framework file for iOS) to be included.
+
+### What is a bindings library
+
+A bindings library is an assembly that wraps the native library with C# wrappers so native code can be invoked via C# calls. Xamarin.Android implements bindings by using Managed Callable Wrappers (MCW) which are a JNI bridge that are used when managed code needs to invoke Java code. In the case of iOS, to bind a library an API definition file is needed; this is merely a C# source file that contains C# interfaces that have been annotated with a handful of attributes that help drive the binding. The API definition is merely the contract that will be used to generate the API.
+
+## Architecture
+
+As can be seen in the diagram below, the Android and iOS bindings libraries make use of the Java and Objective-C code that is already projected from the shared C++ code. The bindings also make use of the native APIs for both platforms yielding a C# library for every platform that can be consumed by any Xamarin platform.
+
+[image](assets/XamarinArchitecture.png)
+
+Something to consider is that Native Xamarin is designed to share the logic and services inside an application but the UI is independent for every platform which allows us to have three different object models.  
+
+## Rendering
+
+### UWP
+
+For Xamarin rendering UWP doesn't require any changes as the rendering and API's are done in C#. The current renderer we provide must only be referenced as is or included as a nuget package.
+
+### Android
+
+Android bindings are mostly done in the Metadata.xml file which defines the contract between C# and Java. This file can contain renames for packages, change of return types and parameter types and names as well as removing classes or including extra classes.
+
+``` xml
+<metadata>
+  <attr path="/api/package[@name='io.adaptivecards']" name="managedName">AdaptiveCards.Rendering.Xamarin.Android</attr>
+
+  <!-- Updates Get methods to return java.lang.Object instead of the actual type -->
+  <attr path="/api/package[@name='io.adaptivecards.objectmodel']/class[@name='BaseActionElementVector']/ method[@name='get']" name="managedReturn">Java.Lang.Object</attr>
+
+  <!-- Updates DoInBackground method's parameter and return types -->
```

Comment:
[Team Spec Review] Talk about the P0's etc upfront here rather than at the bottom to put some context around the definitions referenced below; (tabular form might be best and more reader-friendly)